src/rpc/rawtransaction.cpp

```diff
@@ -933,6 +951,18 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
             result_0.pushKV("reject-reason", state.GetRejectReason());
         }
     }
+    // Calculate and push tx fee, only if test_accept_res == true
```

Comment:
No need to calculate this again. `AcceptToMemoryPool` already did this calculation. It should be possible for `AcceptToMemoryPool` to simply pass out the fee.

Reply:
Since the lock is released, I am not sure if this is even safe, since coins could vanish?!

Reply:
See also https://github.com/bitcoin/bitcoin/pull/15810#issuecomment-482728862

Reply:
Yes. Will try that. That went scarily into validation section of the protocol. So skipped that as an initial trial.

Reply:
Quick question before I go on doing that. 
It seems the way to extract fee is by changing the signature of `AcceptToMemoryPool` and passing a `CAmount&` argument. But this function is being called in many other (nontrivial) places. So all of them need to be modified, and most of them don't even need to get back the fee. I am afraid it can be an overkill (security bug?) to get a feature.

Is this the only way to do it, or there are other methods possible?

Reply:
It is possible to add optional return parameters by passing in a pointer. If the pointer is NULL (the default), it will be ignored. If the pointer points to something the return value will be written there.
Thus no caller will need to be updated because the return value is optional.


Example draft:

```cpp
bool AcceptToMemoryPool(CTxMemPool& pool, TxValidationState &state, const CTransactionRef &tx,
                        std::list<CTransactionRef>* plTxnReplaced,
                        bool bypass_limits, const CAmount nAbsurdFee,
                        CAmount* fee_out=nullptr, bool test_accept=false) EXCLUSIVE_LOCKS_REQUIRED(cs_main);